### PR TITLE
Add compact-mode 

### PR DESCRIPTION
**Pull request by Alirezanet**
We had a lot of problems with VGT on mobile devices. so I decided to add a mobile-friendly feature to VGT.

---

*Normal Mode:*
image

---
*Compact Mode:*

please approve pull-request if you like it.

### CODE REVIEW

**Comment by Alirezanet** [comment]
should I include the dist folder in the pull request?

File: src/components/Table.vue [hunk 1]
```diff
@@ -341,6 +342,7 @@ export default {
     responsive: { default: true },
     rtl: { default: false },
     rowStyleClass: { default: null, type: [Function, String] },
+    compactMode: { default: false, type: Boolean},
```
Comment (p0psicles) [inline src/components/Table.vue:345]: Properties that you pass as an attribute that are boolean and default false, can just be specified like: `compactMode: Boolean`. Then you can pass it like an attribute without value. Like: `<component compact-mode />`

But I'm not the dev. 
Very nice feature though!

**Comment by p0psicles** [comment]
I also think you should update the docs in vp-docs

File: src/components/Table.vue (hunk 1 above)
Comment (Alirezanet) [reply src/components/Table.vue:345]: sure. thanks for feedback

Comment (Alirezanet) [reply src/components/Table.vue]: added your suggestion.

**Comment by Alirezanet** [comment]
> I also think you should update the docs in vp-docs

I'll update documentation if the maintainer approves this. not sure yet he will

**Comment by Alirezanet** [comment]
> I also think you should update the docs in vp-docs

I've updated vp-docs too.

**Comment by Alirezanet** [comment]
hmm, where is the project maintainer? is this repository still alive?
@xaksis

**Comment by p0psicles** [comment]
It kinda is. But I've also have been Waiting for a review for a couple of months now

**Comment by Alirezanet** [comment]
a couple of months for a review! 
I need this feature in production.  today I had to remove `VGT` and use my own custom one, Although this is not a good idea. what should we do?

**Comment by p0psicles** [comment]
I would recommend to package it your self until merged. That's what I'm doing. Added my fork URL to the package.json. but I'm just running an open source hobby project.

If you have like nexus. Just publish it your self.

**Comment by xaksis** [comment]
@Alirezanet @p0psicles thank you for your patience! This is now released as new minor version.

**Comment by Alirezanet** [comment]
thanks.

**Comment by pixeldemon9** [comment]
Not sure if I'm missing something, but compactMode doesn't appear to be working/activating in my app or on the documentation page:
[Docs](https://xaksis.github.io/vue-good-table/guide/configuration/#compactmode)

**Comment by Alirezanet** [comment]
@pixeldemon9 In the documentations you are right, it is not showing compact-mode example, I think documentation is not using the latest version of the component. but if you are using the latest version of VGT on your application you should see the compacted grid in small screen and mobiles (it will be trigger in under 576px screens)

**Comment by pixeldemon9** [comment]
@Alirezanet Discovered I was passing the prop as "compactMode" when it needed to be "compact-mode".  Silly mistake on my part, although I think this is how it is presented in the documentation as well.

Thanks for this PR, I think this is an important option to handle tables on mobile.

**Comment by dashty94** [comment]
The table-column slot does not work in compactMode.

**Comment by Alirezanet** [comment]
> The table-column slot does not work in compactMode.

Ture. this is a design choice  because usually we don't need that slot in compactMode,  
what is your expected behavior ? (because there is not enough space in grid header in compact mode)

**Comment by dashty94** [comment]
> > The table-column slot does not work in compactMode.
> 
> 
> 
> Ture. this is a design choice  because usually we don't need that slot in compactMode,  
> 
> what is your expected behavior ? (because there is not enough space in grid header in compact mode)

Nothing critical. I had to translate the column labels based on the locale but I can also achieve that when setting the columns objects.

**Pull request merged**